include/cadmium/engine/pdevs_dynamic_coordinator.hpp

```diff
@@ -205,7 +227,11 @@ namespace cadmium {
                         LOGGER::template log<cadmium::logger::logger_message_routing, cadmium::logger::coor_routing_eoc_collect>(t, _model_id);
 
                         // Fill all outboxes and clean the inboxes in the lower levels recursively
+                        #ifdef CADMIUM_EXECUTE_CONCURRENT
```

Comment:
You having a lot of "ifdef" for same condition in this class. 
I would say it may be better to add a template parameter for execution_policy to the class with a default to "sequential" and especialize (or use enable if/static_if) for the parallel case. The execution_policy could be the  STL one, https://en.cppreference.com/w/cpp/algorithm/execution_policy_tag